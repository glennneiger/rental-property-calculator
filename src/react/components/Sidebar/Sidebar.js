import * as React from 'react';
import PropTypes from 'prop-types';

const css = require('./sidebar.css');
import GuestSidebar from '../GuestSidebar';
import UserSidebar from '../UserSidebar';

const Sidebar = ({
  auth
}) => {
  return (
    <div className={css.sidebar}>
      {auth.isAuthenticated
        ? <UserSidebar />
        : <GuestSidebar />
      }
    </div>
  );
};

// Sidebar.propTypes = {
//   auth: PropTypes.object.isRequired
// };

export default Sidebar;