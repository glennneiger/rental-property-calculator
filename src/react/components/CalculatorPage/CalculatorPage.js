import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { ToastContainer, toast } from 'react-toastify';

import Calculator from '../Calculator';
import Sidebar from '../Sidebar';
import css from './calculatorPage.css';
import { SIDEBAR_MAX_SCREEN_WIDTH } from '../../../constants';

class CalculatorPage extends Component {
  componentDidMount() {
    window.addEventListener('beforeunload', this.handleUnload);
  }
  componentWillUnmount() {
    window.removeEventListener('beforeunload', this.handleUnload);
  }
  handleUnload = e => {
    if (this.props.changesMade) {
      e.preventDefault();
      e.returnValue = null;
    }
  }
  render() {
    return (
      <div className={css.calculatorPage}>
        {this.props.sidebarVisible
          ? <Sidebar />
          : null
        }
        {this.props.screenWidth < SIDEBAR_MAX_SCREEN_WIDTH
          && this.props.sidebarVisible
          ? null
          : <Calculator />
        }
        <ToastContainer autoClose={3000} position={toast.POSITION.BOTTOM_RIGHT} />
      </div>
    );
  }
}

CalculatorPage.propTypes = {
  changesMade: PropTypes.bool.isRequired,
  screenWidth: PropTypes.number.isRequired,
  sidebarVisible: PropTypes.bool.isRequired
};

export default CalculatorPage;