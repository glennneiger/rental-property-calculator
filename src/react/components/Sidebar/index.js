import { connect } from 'react-redux';

import Sidebar from './Sidebar';

const mapStateToProps = state => ({
  auth: state.auth
});

export default connect(
  mapStateToProps
)(Sidebar);