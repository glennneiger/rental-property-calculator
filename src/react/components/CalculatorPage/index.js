import { connect } from 'react-redux';

import CalculatorPage from './CalculatorPage';

const mapStateToProps = state => ({
  screenWidth: state.ui.screenWidth,
  sidebarVisible: state.sidebarVisible
});

export default connect(
  mapStateToProps
)(CalculatorPage);