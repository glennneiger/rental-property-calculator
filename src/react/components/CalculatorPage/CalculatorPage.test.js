import React from 'react';
import { shallow } from 'enzyme';
import { expect as chaiExpect } from 'chai';

import CalculatorPage from './CalculatorPage';
import Sidebar from '../Sidebar';

describe('<CalculatorPage />', () => {
  const mockWideScreenWidth = 1920;
  const mockNarrowScreenWidth = 500;

  it('has proper className for styling', () => {
    const wrapper = shallow(<CalculatorPage sidebarVisible={true} />);

    chaiExpect(wrapper).to.have.className('calculatorPage');
  });

  it('renders Sidebar when sidebarVisible is true', () => {
    const wrapper = shallow(
      <CalculatorPage screenWidth={mockWideScreenWidth} sidebarVisible={true} />
    );

    chaiExpect(wrapper.find(Sidebar)).to.have.length(1);
  });

  it('renders no Sidebar when sidebarVisible is false', () => {
    const wrapper = shallow(
      <CalculatorPage screenWidth={mockWideScreenWidth} sidebarVisible={false} />
    );

    chaiExpect(wrapper.find(Sidebar)).to.have.length(0);
  });

  it('renders Calculator when screenWidth is more than sidebar max', () => {
    const wrapper = shallow(
      <CalculatorPage screenWidth={mockWideScreenWidth} sidebarVisible={true} />
    );

    chaiExpect(wrapper.find('Calculator')).to.have.length(1);
  });

  it('does not render Calculator when screenWidth is less than' +
    ' sidebar max and sidebar is showing', () => {
    const wrapper = shallow(
      <CalculatorPage screenWidth={mockNarrowScreenWidth} sidebarVisible={true} />
    );

    chaiExpect(wrapper.find('Calculator')).to.have.length(0);
  });
});