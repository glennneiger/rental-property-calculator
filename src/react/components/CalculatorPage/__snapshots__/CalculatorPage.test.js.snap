// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CalculatorPage /> renders correctly with Sidebar and Calculator showing 1`] = `
<div
  className="calculatorPage"
>
  <Connect(Sidebar) />
  <Connect(Calculator) />
  <ToastContainer
    autoClose={3000}
    bodyClassName={null}
    className={null}
    closeButton={
      <CloseButton
        ariaLabel="close"
      />
    }
    closeOnClick={true}
    draggable={true}
    draggablePercent={80}
    hideProgressBar={false}
    newestOnTop={false}
    pauseOnFocusLoss={true}
    pauseOnHover={true}
    position="bottom-right"
    progressClassName={null}
    progressStyle={null}
    rtl={false}
    style={null}
    toastClassName={null}
    transition={[Function]}
  />
</div>
`;

exports[`<CalculatorPage /> renders correctly with only Calculator showing 1`] = `
<div
  className="calculatorPage"
>
  <Connect(Calculator) />
  <ToastContainer
    autoClose={3000}
    bodyClassName={null}
    className={null}
    closeButton={
      <CloseButton
        ariaLabel="close"
      />
    }
    closeOnClick={true}
    draggable={true}
    draggablePercent={80}
    hideProgressBar={false}
    newestOnTop={false}
    pauseOnFocusLoss={true}
    pauseOnHover={true}
    position="bottom-right"
    progressClassName={null}
    progressStyle={null}
    rtl={false}
    style={null}
    toastClassName={null}
    transition={[Function]}
  />
</div>
`;

exports[`<CalculatorPage /> renders correctly with only Sidebar showing 1`] = `
<div
  className="calculatorPage"
>
  <Connect(Sidebar) />
  <ToastContainer
    autoClose={3000}
    bodyClassName={null}
    className={null}
    closeButton={
      <CloseButton
        ariaLabel="close"
      />
    }
    closeOnClick={true}
    draggable={true}
    draggablePercent={80}
    hideProgressBar={false}
    newestOnTop={false}
    pauseOnFocusLoss={true}
    pauseOnHover={true}
    position="bottom-right"
    progressClassName={null}
    progressStyle={null}
    rtl={false}
    style={null}
    toastClassName={null}
    transition={[Function]}
  />
</div>
`;
